packages/sdk/react-surface/src/Plugin.ts

```diff
@@ -4,37 +4,92 @@
 
 import { FC, PropsWithChildren } from 'react';
 
+/**
+ * Props passed to a component by the `Surface` resolver.
+ */
 export type ComponentProps = PropsWithChildren<{
   data: any;
   role?: string;
 }>;
 
 export type PluginProvides<TProvides> = TProvides & {
+  /**
+   * React Context which is wrapped around the application to enable any hooks the plugin may provide.
+   */
   context?: FC<PropsWithChildren>;
+
+  /**
+   * Used by the `Surface` resolver to find a component to render when presented with data but no component name.
+   */
   component?: <P extends PropsWithChildren = PropsWithChildren>(
     data: unknown,
     role?: string,
     props?: Partial<P>,
   ) => FC<ComponentProps> | undefined | null | false | 0;
+
+  /**
+   * Used by the `Surface` resolver to find a component by name.
+   */
   components?: Record<string, FC<ComponentProps>> & { default?: FC };
 };
 
 export type Plugin<TProvides = {}> = {
```

Comment:
> Applications consistent entirely from a set of Plugins that provide different capabilities, such as UI components, data sources, and other application infrastructure.

(Top level type comments -- esp. for these key concepts -- are the most important -- and often most difficult to write).